src/main/java/org/team199/wpiws/connection/MessageProcessor.java

```diff
@@ -70,14 +80,37 @@ public static void registerProcessor(String type, DeviceMessageProcessor process
      * @param data the values of that device which have been modified
      */
     public static void process(String device, String type, List<WSValue> data) {
-        DeviceMessageProcessor processor = processors.get(type);
+        // Per the spec, some message with a type of SimDevice have a data
+        // format that is identical to hardware devices. In particular a
+        // SimDevice message with device=DutyCyle:Name has the same data format
+        // as a DutyCycle message, and a SimDevice message with
+        // device=CAN{Gyro,AI,Accel,DIO,DutyCycle}:Name has the same data format
+        // as a {Gyro,AI,Accel,DIO,DutyCycle} message. The
+        // CAN{Gyro,AI,Accel,DIO,DutyCycle} processors are registered as aliases
+        // for the the their non CAN counterparts. CANMotor and CANEncoder have
+        // their own processors because there is no Motor processor and the
+        // CANEncoder data format is different from the Encoder data format.
+        String dataType = type;
+        if (type.equals("SimDevice")) {
+            String[] deviceParts = device.split(":");
+            if (deviceParts.length == 2) {
```

Comment:
Support can device names containing `":"`.

```suggestion
            if (deviceParts.length > 1) {
```